draft-ietf-quic-tls.md

```diff
@@ -505,26 +508,24 @@ Rekey tx to 0-RTT Keys
                                                    Get Handshake
                      <------------- Initial
                                           Rekey rx to 0-RTT keys
-                                              Handshake Received
-                                      Rekey rx to Handshake keys
+                                      Rekey tx to Handshake keys
                                                    Get Handshake
                      <----------- Handshake
+                                      Rekey rx to Handshake keys
                                           Rekey tx to 1-RTT keys
                      <--------------- 1-RTT
 Handshake Received
-Rekey rx to Handshake keys
+Rekey tx and rx to Handshake keys
 Handshake Received
 Get Handshake
 Handshake Complete
                      Handshake ----------->
-Rekey tx to 1-RTT keys
+Rekey tx and rx to 1-RTT keys
                      1-RTT --------------->
                                               Handshake Received
                                           Rekey rx to 1-RTT keys
-                                                   Get Handshake
```

Comment:
This bit (and the next) need to remain for NewSessionTicket.

Do you think that it would help to identify which TLS messages are being sent and received here?

Reply:
I would have said no, except for the bit about the NST. I might file another PR about that. The kind of ugly thing about this diagram is rekeying the rx from 0RTT to Handshake. There's no actual trigger for this; in practice we're actually keeping both keys, of course.

Reply:
Perhaps it would be more appropriate to say "Install Handshake rx keys," then?  We have the separate PR forthcoming about when keys from previous levels are dropped; perhaps in this diagram we could exclusively talk about installing them.